internal/operands/common-instancetypes/reconcile.go

```diff
@@ -38,16 +49,45 @@ type clusterType interface {
 	instancetypev1alpha2.VirtualMachineClusterInstancetype | instancetypev1alpha2.VirtualMachineClusterPreference
 }
 
-func fetchClusterResources[C clusterType](bundlePath string) ([]C, error) {
-	file, err := ioutil.ReadFile(bundlePath)
-	if err != nil {
-		return nil, err
+func (c *commonInstancetypes) Name() string {
+	return operandName
+}
+
+func WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &instancetypev1alpha2.VirtualMachineClusterInstancetype{}, Crd: instancetypeapi.ClusterPluralResourceName, WatchFullObject: true},
+		{Object: &instancetypev1alpha2.VirtualMachineClusterPreference{}, Crd: instancetypeapi.ClusterPluralPreferenceResourceName, WatchFullObject: true},
 	}
-	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+}
+
+func (c *commonInstancetypes) WatchClusterTypes() []operands.WatchType {
+	return WatchClusterTypes()
+}
+
+func (c *commonInstancetypes) WatchTypes() []operands.WatchType {
+	return nil
+}
+
+func (c *commonInstancetypes) RequiredCrds() []string {
+	return []string{
+		virtualMachineClusterInstancetypeCrd,
+		virtualMachineClusterPreferenceCrd,
+	}
+}
+
+func New(virtualMachineClusterInstancetypeBundlePath, virtualMachineClusterPreferenceBundlePath string) *commonInstancetypes {
+	return &commonInstancetypes{
+		virtualMachineClusterInstancetypeBundle: virtualMachineClusterInstancetypeBundlePath,
+		virtualMachineClusterPreferenceBundle:   virtualMachineClusterPreferenceBundlePath,
+	}
+}
+
+func decodeResources[C clusterType](b []byte) ([]C, error) {
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(b), 1024)
```

Comment:
I asked the same question in `vm-console-proxy`, do we want to support both YAML and JSON as input formats?

Reply:
The support for JSON is incidental here, we want the features provided by the decoder over the pure YAML reader. Mainly that it decodes per object instead of the entire YAML document.

Reply:
Can you use `NewYAMLToJSONDecoder` instead? It should have the desired behavior while only allowing YAML.

Reply:
I would say yes for both formats. In `vm-console-proxy` it was special case, because the config file was yaml. But here, we would need to unnecessarily translate json to yaml when encoding, and then back from yaml to json when decoding.